Lib/json/tool.py

```diff
@@ -14,38 +14,69 @@
 import json
 import sys
 
+def _read(infile, json_lines):
+            try:
+                if json_lines:
+                    return (json.loads(line) for line in infile)
+                else:
+                    return (json.load(infile), )
+            except ValueError as e:
+                raise SystemExit(e)
+
+def _open_outfile(outfile, parser):
+    try:
+        return sys.stdout if outfile == '-' else open(outfile, 'w')
```

Comment:
`FileType` can automatically manage this use case `'-'` and return `sys.stdin/sys.stdout` based on the open mode.

Prefer to use it.
https://docs.python.org/3/library/argparse.html#argparse.FileType